### PR TITLE
Fix #494 - cleanup vendor bundle

### LINKED ISSUE #494: Clean up the vendor package by adding a .gitattributes
You distribute useless files to users, a `.gitattributes` files will fix it.
`du -sh vendor/mangopay/php-sdk-v2/` => `1,6M`

```
vendor/mangopay/php-sdk-v2/
├── CHANGELOG.md
├── composer.json
├── demos
│   ├── api
│   │   ├── config.php
│   │   ├── demo.css
│   │   ├── demo.js
│   │   ├── form.php
│   │   ├── htmlHelper.php
│   │   ├── index.php
│   │   └── menu.php
│   ├── paymentDirect
│   │   ├── config.php
│   │   ├── index.php
│   │   ├── js
│   │   │   └── script.js
│   │   ├── non_js.php
│   │   ├── payment.php
│   │   └── with_js.php
│   ├── transaction_report_for_cron.php
│   ├── users_all.php
│   ├── users_create.php
│   ├── users_get.php
│   └── workflow
│       ├── inc
│       │   ├── config.php
│       │   └── mockStorage.php
│       ├── index.php
│       ├── logo.png
│       ├── scripts
│       │   ├── bankaccount.php
│       │   ├── card-reg.php
│       │   ├── card-reg-put.php
│       │   ├── client.php
│       │   ├── end.php
│       │   ├── intro.php
│       │   ├── kyc.php
│       │   ├── payin-applepay-direct.php
│       │   ├── payin-card-direct.php
│       │   ├── payin-card-preauth.php
│       │   ├── payin-card-web.php
│       │   ├── payin-card-web-review.php
│       │   ├── payin-googlepay-direct.php
│       │   ├── payout.php
│       │   ├── preauth.php
│       │   ├── refund-payin.php
│       │   ├── refund-transfer.php
│       │   ├── transfer.php
│       │   ├── user-create-legal.php
│       │   ├── user-create-natural.php
│       │   ├── wallet-create2.php
│       │   └── wallet-create.php
│       ├── style.css
│       └── syntaxhighlighter
│           ├── shBrushPhp.js
│           ├── shCore.js
│           └── shThemeDefault.css
├── LICENSE
├── Makefile
├── MangoPay
│   ├── Address.php
│   ├── ApiBankAccounts.php
│   ├── ApiBankingAliases.php
│   ├── ApiCardPreAuthorizations.php
│   ├── ApiCardRegistrations.php
│   ├── ApiCards.php
│   ├── ApiClients.php
│   ├── ApiDisputeDocuments.php
│   ├── ApiDisputes.php
│   ├── ApiEvents.php
│   ├── ApiHooks.php
│   ├── ApiKycDocuments.php
│   ├── ApiMandates.php
│   ├── ApiPayIns.php
│   ├── ApiPayOuts.php
│   ├── ApiRefunds.php
│   ├── ApiReports.php
│   ├── ApiRepudiations.php
│   ├── ApiResponses.php
│   ├── ApiTransfers.php
│   ├── ApiUboDeclarations.php
│   ├── ApiUsers.php
│   ├── ApiWallets.php
│   ├── Autoloader.php
│   ├── AVSResult.php
│   ├── BankAccountDetailsCA.php
│   ├── BankAccountDetailsGB.php
│   ├── BankAccountDetailsIBAN.php
│   ├── BankAccountDetailsOTHER.php
│   ├── BankAccountDetails.php
│   ├── BankAccountDetailsUS.php
│   ├── BankAccount.php
│   ├── BankingAliasIBAN.php
│   ├── BankingAliasOTHER.php
│   ├── BankingAlias.php
│   ├── BankingAliasType.php
│   ├── Billing.php
│   ├── Birthplace.php
│   ├── BrowserInfo.php
│   ├── BusinessType.php
│   ├── Card.php
│   ├── CardPreAuthorizationPaymentStatus.php
│   ├── CardPreAuthorization.php
│   ├── CardPreAuthorizationStatus.php
│   ├── CardRegistration.php
│   ├── CardRegistrationStatus.php
│   ├── CardValidity.php
│   ├── ClientLogoUpload.php
│   ├── Client.php
│   ├── CurrencyIso.php
│   ├── DebitedBankAccount.php
│   ├── DeclaredUbo.php
│   ├── DeclaredUboStatus.php
│   ├── DisputeDocumentPage.php
│   ├── DisputeDocument.php
│   ├── DisputeDocumentStatus.php
│   ├── DisputeDocumentType.php
│   ├── Dispute.php
│   ├── DisputeReason.php
│   ├── DisputeReasonType.php
│   ├── DisputeStatus.php
│   ├── DisputeType.php
│   ├── DocumentPageConsult.php
│   ├── EMoney.php
│   ├── Event.php
│   ├── EventType.php
│   ├── FilterBankAccounts.php
│   ├── FilterBase.php
│   ├── FilterCards.php
│   ├── FilterDisputeDocuments.php
│   ├── FilterDisputes.php
│   ├── FilterEvents.php
│   ├── FilterKycDocuments.php
│   ├── FilterPreAuthorizations.php
│   ├── FilterRefunds.php
│   ├── FilterReports.php
│   ├── FilterTransactions.php
│   ├── FundsType.php
│   ├── Hook.php
│   ├── HookStatus.php
│   ├── HookValidity.php
│   ├── KeyValueArray.php
│   ├── KycDocument.php
│   ├── KycDocumentRefusedReasonType.php
│   ├── KycDocumentStatus.php
│   ├── KycDocumentType.php
│   ├── KycLevel.php
│   ├── KycPage.php
│   ├── LegalPersonType.php
│   ├── Libraries
│   │   ├── ApiBase.php
│   │   ├── ApiOAuth.php
│   │   ├── AuthenticationHelper.php
│   │   ├── AuthorizationTokenManager.php
│   │   ├── Configuration.php
│   │   ├── DefaultStorageStrategy.php
│   │   ├── Document.php
│   │   ├── Dto.php
│   │   ├── EntityBase.php
│   │   ├── Error.php
│   │   ├── Exception.php
│   │   ├── HttpBase.php
│   │   ├── HttpCurl.php
│   │   ├── HttpResponse.php
│   │   ├── IStorageStrategy.php
│   │   ├── Logs.php
│   │   ├── OAuthToken.php
│   │   ├── RequestType.php
│   │   ├── ResponseException.php
│   │   ├── RestTool.php
│   │   ├── Upload.php
│   │   └── UrlTool.php
│   ├── Mandate.php
│   ├── MandateStatus.php
│   ├── MangoPayApi.php
│   ├── Money.php
│   ├── NaturalUserCapacity.php
│   ├── Pagination.php
│   ├── PayInCardTemplateURLOptions.php
│   ├── PayInExecutionDetailsDirect.php
│   ├── PayInExecutionDetailsExternalInstruction.php
│   ├── PayInExecutionDetails.php
│   ├── PayInExecutionDetailsWeb.php
│   ├── PayInExecutionType.php
│   ├── PayInPaymentDetailsApplePay.php
│   ├── PayInPaymentDetailsBankWire.php
│   ├── PayInPaymentDetailsCard.php
│   ├── PayInPaymentDetailsDirectDebitDirect.php
│   ├── PayInPaymentDetailsDirectDebit.php
│   ├── PayInPaymentDetailsGooglePay.php
│   ├── PayInPaymentDetailsPaypal.php
│   ├── PayInPaymentDetails.php
│   ├── PayInPaymentDetailsPreAuthorized.php
│   ├── PayInPaymentType.php
│   ├── PayIn.php
│   ├── PayInRecurring.php
│   ├── PayInRecurringRegistration.php
│   ├── PayInRecurringRegistrationRequestResponse.php
│   ├── PayInStatus.php
│   ├── PayInTemplateURLOptions.php
│   ├── PayInWebExtendedView.php
│   ├── PaymentData.php
│   ├── PayOutPaymentDetailsBankWire.php
│   ├── PayOutPaymentDetails.php
│   ├── PayOutPaymentType.php
│   ├── PayOut.php
│   ├── PayOutStatus.php
│   ├── PersonType.php
│   ├── PlatformCategorization.php
│   ├── RateLimit.php
│   ├── RecurringPayInCIT.php
│   ├── RecurringPayInMIT.php
│   ├── Refund.php
│   ├── RefundReasonDetails.php
│   ├── RefundStatus.php
│   ├── ReportRequest.php
│   ├── ReportStatus.php
│   ├── ReportType.php
│   ├── Repudiation.php
│   ├── Response.php
│   ├── ScopeBlocked.php
│   ├── Sector.php
│   ├── SecurityInfo.php
│   ├── SettlementTransfer.php
│   ├── ShippingAddress.php
│   ├── Shipping.php
│   ├── SortDirection.php
│   ├── Sorting.php
│   ├── TransactionNature.php
│   ├── Transaction.php
│   ├── TransactionStatus.php
│   ├── TransactionType.php
│   ├── Transfer.php
│   ├── UboDeclaration.php
│   ├── UboDeclarationRefusedOrIncompleteReasonType.php
│   ├── UboDeclarationRefusedReasonType.php
│   ├── UboDeclarationStatus.php
│   ├── Ubo.php
│   ├── UboRefusedReasonType.php
│   ├── UserBlockStatus.php
│   ├── UserLegal.php
│   ├── UserNatural.php
│   ├── User.php
│   └── Wallet.php
├── php_env
│   ├── PHP_5.6
│   │   ├── composer.sh
│   │   └── Dockerfile
│   ├── PHP_7.0
│   │   ├── composer.sh
│   │   └── Dockerfile
│   └── PHP_8.0
│       ├── composer.sh
│       └── Dockerfile
├── phpunit.xml
├── README.md
└── tests
    ├── Cases
    │   ├── BankAccountsTest.php
    │   ├── BankingAliasTest.php
    │   ├── Base.php
    │   ├── CardPreAuthorizationsTest.php
    │   ├── CardRegistrationsTest.php
    │   ├── CardsTest.php
    │   ├── ClientTest.php
    │   ├── ConfigurationTest.php
    │   ├── DisputesTest.php
    │   ├── ErrorTests.php
    │   ├── EventsTest.php
    │   ├── HooksTest.php
    │   ├── IdempotencyTest.php
    │   ├── KycDocumentsTest.php
    │   ├── MandatesTest.php
    │   ├── PayInsTest.php
    │   ├── PayOutsTest.php
    │   ├── RateLimitTest.php
    │   ├── RefundsTest.php
    │   ├── ReportsTest.php
    │   ├── TokensTest.php
    │   ├── TransfersTest.php
    │   ├── UboDeclarationsTest.php
    │   ├── UsersTest.php
    │   └── WalletsTest.php
    ├── Mocks
    │   └── MockStorageStrategy.php
    └── TestKycPageFile.png

17 directories, 269 files
```

### PR DESCRIPTION
**Pull request by williamdes**
Use `git archive HEAD --format zip --output=repo.zip` if you want to test.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+.travis.yml export-ignore
+# Not existing for now
+.github export-ignore
+.editorconfig export-ignore
+
+# Dev & test files
+Makefile export-ignore
+/demos export-ignore
+/tests export-ignore
```
Comment (williamdes) [inline .gitattributes:12]: Agree to remove those, all the other ones are no needed files by anyone

**Comment by williamdes** [comment]
Hi @fredericdelordm 
Did you talk about something you did not merge in the release notes ^^ ?

**Comment by SoloJr** [comment]
Hi @williamdes 
What is it about? If so, it’s my mistake, not his. 

SoloJe

**Comment by williamdes** [comment]
> Hi @williamdes
> What is it about? If so, it’s my mistake, not his.
> 
> SoloJr

On https://github.com/Mangopay/mangopay2-php-sdk/releases/tag/3.7.0 it says "Cleanup the vendor bundle, to reduce the size of the SDK"

But my work was not merged, the release notes **and** changelog need to be edited

File: .gitattributes [hunk 2]
```diff
@@ -0,0 +1,11 @@
+.gitattributes export-ignore
+.gitignore export-ignore
+
+.github export-ignore
+.editorconfig export-ignore
+
+# Dev & test files
+Makefile export-ignore
+/php_env export-ignore
+phpunit.xml export-ignore
```
Comment (williamdes) [inline .gitattributes:10]: This one should be removed or the tests directory be added back
I missed that

Anyway I doubt anyone runs tests using the composer package

**Pull request merged**